src/PackagePrivate/Humanizer/InternationalizedHumanizer.php

```diff
@@ -64,9 +64,16 @@ public function __construct( MessageBuilder $messageBuilder, LanguageStrategy $l
         $this->languageStrategy = $languageStrategy;
     }
 
-	public function humanize( EdtfValue $edtf ): string {
+    /**
+     * @param EdtfValue $edtf
+     * @param EdtfValue|null $context Having the context means that value is going to be humanized in context of
+     * other EDTF value. I.e. date is being humanized in context of interval value. Sometimes this context can be
+     * useful to determine additional details for date humanization
+     * @return string
+     */
+	public function humanize( EdtfValue $edtf, ?EdtfValue $context = null ): string {
```

Comment:
That is a lot of added complexity :/ Looks like you can simplify it a bit by having a boolean parameter instead. Still, would be nice if we could avoid adding an extra parameter to all these functions.

Reply:
TBH I'd just drop this code. Not sure anyone minds if the year is always lowercase or uppercase.

Another way of addressing the context is in the code that deals with context. It could call ucfirst on the whole value.